diff --git a/drivers/net/wireless/ath/regd.c b/drivers/net/wireless/ath/regd.c
index f15e7bd690b5..b948ebcf2d60 100644
--- a/drivers/net/wireless/ath/regd.c
+++ b/drivers/net/wireless/ath/regd.c
@@ -61,6 +61,9 @@ static int __ath_regd_init(struct ath_regulatory *reg);
 #define ATH_5GHZ_NO_MIDBAND	ATH_5GHZ_5150_5350, \
 				ATH_5GHZ_5725_5850
 
+/* This can be used to override EEPROM factory burned country */
+#define ATH_USER_COUNTRY <CountryCode to use>
+
 /* Can be used for:
  * 0x60, 0x61, 0x62 */
 static const struct ieee80211_regdomain ath_world_regdom_60_61_62 = {
@@ -428,8 +431,7 @@ static void ath_reg_apply_radar_flags(struct wiphy *wiphy,
 		 *   do not allow AP mode
 		 */
 		if (!(ch->flags & IEEE80211_CHAN_DISABLED))
-			ch->flags |= IEEE80211_CHAN_RADAR |
-				     IEEE80211_CHAN_NO_IR;
+			ch->flags |= IEEE80211_CHAN_RADAR;
 	}
 }
 
@@ -690,6 +692,11 @@ static int __ath_regd_init(struct ath_regulatory *reg)
 
 	ath_regd_sanitize(reg);
 
+#if defined(ATH_USER_COUNTRY)
+	printk(KERN_DEBUG "ath: EEPROM regdomain 0x%0x overwritten with 0x%0x\n", reg->current_rd, COUNTRY_ERD_FLAG | ATH_USER_COUNTRY);
+	reg->current_rd = COUNTRY_ERD_FLAG | ATH_USER_COUNTRY;
+#endif
+
 	printk(KERN_DEBUG "ath: EEPROM regdomain: 0x%0x\n", reg->current_rd);
 
 	if (!ath_regd_is_eeprom_valid(reg)) {
